test/e2e_node/image_list.go

```diff
@@ -70,6 +70,11 @@ var NodePrePullImageList = sets.NewString(
 // 2. the ones passed in from framework.TestContext.ExtraEnvs
 // So this function needs to be called after the extra envs are applied.
 func updateImageAllowList(ctx context.Context) {
+	// Architecture-specific image
+	if !utils.IsRunningOnArm64() {
```

Comment:
The e2e binary can be running against a remote host, right? This doesn't seem like the correct check.

Reply:
My current testing is with `run_remote` where I trigger the test from x86 machine to execute test on the ARM64 machine. I think for current test-infra, this is how the e2e_node would be executed.

```
go run /usr/local/google/home/ikema/go-k8s-oss/src/k8s.io/kubernetes/test/e2e_node/runner/remote/run_remote.go --cleanup -vmodule=*=4 --ssh-env=gce --results-dir=/tmp/e2e-node-results/ason --project=ikema-gke-dev-2 --use-dockerized-build=true --target-build-arch=arm64 --zone=us-central1-a --ssh-user=ikema --ssh-key=/usr/local/google/home/ikema/.ssh/google_compute_engine --ginkgo-flags='--nodes=1 --focus="\[Serial\]" --skip="\[Flaky\]|\[Slow\]|\[Benchmark\]|\[NodeSpecialFeature:.+\]|\[NodeSpecialFeature\]|\[NodeAlphaFeature:.+\]|\[NodeAlphaFeature\]|\[NodeFeature:Eviction\]|\[NodeFeature:NodeProblemDetector\]|\[NodeFeature:OOMScoreAdj\]|\[NodeFeature:DevicePluginProbe\]|\[NodeConformance\]" '  --test-timeout=5h0m0s --image-config-file=/tmp/e2e-node-results/ason/image-config.yaml 
```

I'm not too familiar with the case of executing e2e binary against a remote host. Do you have any suggestion on what's the most effective check for that case ?

Reply:
that seems right, but then this logic is confusing, because it's the local process architecture.

I had to reread the remote runner to understand the e2e binary actually executes on the remote host.

for cluster e2es we usually just talk to the remote machine remotely and we've had broken assumptions versus the remote machines before, I over pattern matched on that :-)